step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/execution/RepositoryWithAutomationPackageSupport.java

```diff
@@ -354,24 +354,24 @@ public AutomationPackage getAutomationPackage() {
             return getInMemoryManager().getAllAutomationPackages(null).findFirst().orElse(null);
         }
 
-        public boolean isExternallyCreatedContext() {
-            return externallyCreatedContext;
+        public boolean isShared() {
+            return shared;
         }
 
         @Override
         public void close() throws IOException {
-
-            // only after isolated execution is finished we can clean up temporary created resources
-            try {
-                // remove the context from isolated automation package repository
-                log.info("Cleanup isolated execution context");
-
-                IsolatedAutomationPackageRepository.PackageExecutionContext automationPackageManager = sharedPackageExecutionContexts.get(contextId);
+            // remove the context from isolated automation package repository
+            log.info("Cleanup isolated execution context");
+            //In case the Package execution context is shared (i.e. when triggering isolated executions from CLI), we close the shared context
+            //and remove it from the shared map
+            if (shared) {
+                IsolatedAutomationPackageRepository.PackageExecutionContext automationPackageManager = sharedPackageExecutionContexts.remove(contextId);
                 if (automationPackageManager != null) {
                     automationPackageManager.getInMemoryManager().cleanup();
                 }
-            } finally {
-                sharedPackageExecutionContexts.remove(contextId);
+            //Otherwise directly clean the automation package stored in this context
+            } else {
+                inMemoryManager.cleanup();
```

Comment:
Not removing the context from the map anymore in this case. Is this intended?

Reply:
It is now removed on line 368 directly, no reason to remove it in a finally, the map is a concurrentHashMap